app/src/screen.c

```diff
@@ -454,6 +466,90 @@ update_texture(struct screen *screen, const AVFrame *frame) {
     }
 }
 
+bool
+screen_render_text(struct screen *screen) {
+    struct render_text_request req;
+
+    // we check this flag outside lock, it may cause that we don't show text immediately.
```

Comment:
(but it's racy, you should use an atomic instead)

Reply:
Yes, it will race. But the effect is just that it will delay notice that it needs to show text.

This trick added only because I notice the performance has some degrade. Maybe if we don't create texture every frame, we don't need this.